src/test/java/snah/util/ParserTest.java

```diff
@@ -0,0 +1,60 @@
+package snah.util;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
+    @Test
+    public void getEventPayloadTest() {
```

Comment:
Good use of logical blocks for the different test cases but maybe you could separate out the different conditions into different test and name it according to the three part format